src/screens/RecipesList/useRecipesListController.ts

```diff
@@ -32,26 +32,26 @@ export const useRecipeListController = () => {
   useEffect(() => {
     setLoading(true);
 
-    Promise.all([getRecipes(searchOptions), getCardType()])
+    Promise.all([getRecipes({ searchTerm: searchOptions.searchTerm, ...searchOptions.pendingOptions }), getCardType()])
       .then(() => setLoading(false));
-  }, [searchOptions.sort, searchOptions.searchTerm]);
+  }, [searchOptions.pendingOptions.sort, searchOptions.searchTerm]);
 
   useEffect(() => {
     if (!recipes.length) {
       return;
     }
 
-    updateFilter(searchOptions);
-  }, [searchOptions.filter]);
+    updateFilter({ searchTerm: searchOptions.searchTerm, ...searchOptions.pendingOptions });
```

Comment:
earchOptions.pendingOptions can contain searchTerm.
You wrote the code in a way that "searchTerm: searchOptions.searchTerm" will be overwritten by "...searchOptions.pendingOptions"